### DISCUSSION TITLE
set variables from file

### DISCUSSION THREAD

**Discussion by coaperator** [dokku/dokku#4928] [Q&A]
hi, i have three branches test, dev, prod
I need to set environment variables when deploying
how can i do this from a file .env-test, .env-dev, .env-prod ?

**Comment by josegonzalez** [comment]
You cannot set environment variables from a file _during_ a deploy, but you may be able to set them en-masse via the `dokku config:set` command. Not knowing the format of your files, you can try something like:

```shell
# the contents of the .env file:
KEY=VALUE
SOME_OTHER_KEY=VALUE2

# where .env is a file on the remote dokku server and you are running the dokku command on the server
dokku config:set $APP $(cat .env | tr "\n" " ")
```

**Comment by AdrianPop** [comment]
I'm using this: `cat appname.env | xargs dok config:set appname`

**Reply by tombohub** [reply]
does that mean you have .env file on server? How to do it from local laptop?
